src/main/java/com/redhat/devtools/lsp4ij/settings/jsonSchema/MyLightVirtualFile.java

```diff
@@ -0,0 +1,187 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.redhat.devtools.lsp4ij.settings.jsonSchema;
+
+import com.intellij.lang.Language;
+import com.intellij.openapi.fileTypes.CharsetUtil;
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.fileTypes.FileTypeRegistry;
+import com.intellij.openapi.util.NlsSafe;
+import com.intellij.openapi.util.io.FileTooBigException;
+import com.intellij.openapi.util.io.FileUtilRt;
+import com.intellij.openapi.vfs.VfsUtilCore;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.limits.FileSizeLimit;
+import com.intellij.testFramework.LightVirtualFileBase;
+import com.intellij.util.ArrayUtil;
+import com.intellij.util.LocalTimeCounter;
+import com.intellij.util.ThreeState;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * In-memory implementation of {@link VirtualFile}.
+ */
+public class MyLightVirtualFile extends LightVirtualFileBase {

```

Comment:
This looks like a clone of `LightVirtualFile` for the most part. Should it just extend `LightVirtualFile` instead of `LightVirtualFileBase` and override things selectively? Or is that impossible and this has to be almost a clone of `LightVirtualFile` to accomplish what you need?

Reply:
Exactlyit is a copy/paste from LightVirtualFile. I did that because of this line code https://github.com/JetBrains/intellij-community/blob/f873c09975f3262558093855d04474598356d2a5/json/src/com/jetbrains/jsonSchema/impl/light/nodes/JsonSchemaObjectStorage.kt#L74 which doesn't take care of the modificationStamp of the LightVirtualFile (in 2022.3 t is working with LightVirtualFile  but not with 2024.3)

Reply:
Add a comment about that.